gems/pending/util/duplicate_blocker/dedup_handler.rb

```diff
@@ -63,8 +68,8 @@ def handle(method, *args)
       key = key_generator.call(method, *args)
       desc = descriptor.call(method, *args)
 
-      entry = update_history(key, desc)
-      tripped?(entry) ? on_blocking_call(desc) : method[*args]
+      entry = update_history(key.hash, desc)
+      tripped?(entry, desc) ? on_blocking_call(desc) : method[*args]
```

Comment:
It makes sense to save the description in the entry only when the it is short. The description is also provided by user class so the length is not predictable. For the use of blocking duplicated provider events, I actually pass in the event context which is very lengthy. The use of description is for logging, so it is better to be in detail. 

In a normal situation no duplicate is expected, so every monitored method call, e.g., every event, is unique thus stored as a new hash entry. Every bit of saving is important.